src/common/src/telemetry/report.rs

```diff
@@ -112,3 +126,99 @@ where
     });
     (join_handle, shutdown_tx)
 }
+
+pub fn report_event_common(
+    event_stage: PbTelemetryEventStage,
+    feature_name: String,
+    catalog_id: i64,
+    connector_name: Option<String>,
+    object: Option<PbTelemetryDatabaseObject>,
+    attributes: Option<jsonbb::Value>, // any json string
+    node: String,
+) {
+    let event_tracking_id: String;
+    if let Some(tracking_id) = TELEMETRY_TRACKING_ID.get() {
+        event_tracking_id = tracking_id.to_string();
+    } else {
+        tracing::info!("Telemetry tracking_id is not set, event reporting disabled");
+        return;
+    }
+
+    request_to_telemetry_event(
+        event_tracking_id,
+        event_stage,
+        feature_name,
+        catalog_id,
+        connector_name,
+        object,
+        attributes,
+        node,
+        false,
+    );
+}
+
+fn request_to_telemetry_event(
+    tracking_id: String,
+    event_stage: PbTelemetryEventStage,
+    feature_name: String,
+    catalog_id: i64,
+    connector_name: Option<String>,
+    object: Option<PbTelemetryDatabaseObject>,
+    attributes: Option<jsonbb::Value>, // any json string
+    node: String,
+    is_test: bool,
+) {
+    let event = PbEventMessage {
+        tracking_id,
+        event_time_sec: current_timestamp(),
+        event_stage: event_stage as i32,
+        feature_name,
+        connector_name,
+        object: object.map(|c| c as i32),
+        catalog_id,
+        attributes: attributes.map(|a| a.to_string()),
+        node,
+        is_test,
+    };
+    let report_bytes = event.encode_to_vec();
+
+    tokio::spawn(async move {
+        const TELEMETRY_EVENT_REPORT_TYPE: &str = "event";
+        let url = (TELEMETRY_REPORT_URL.to_owned() + "/" + TELEMETRY_EVENT_REPORT_TYPE).to_owned();
+        post_telemetry_report_pb(&url, report_bytes)
+            .await
+            .unwrap_or_else(|e| tracing::info!("{}", e))
```

Comment:
```suggestion
            .unwrap_or_else(|e| tracing::error!("failed to report telemetry events", error = %e.as_report()))
```

Reply:
I think we should not use `error` level for telemetry, which might make users unhappy. lol (including other existing logs for telemetry, not only here) 

From users' point of view, failing to send telemetry report is indeed nothing harmful.

(Perhaps even `warn` should not be used?)

Reply:
I have the same concern, a user wonders a telemetry's warning can affect the system.

Reply:
Yes, I think we are talking about the same user. lol